session.go

```diff
@@ -487,6 +487,8 @@ func (s *Session) sendMsg(hdr header, body []byte, deadline <-chan struct{}) err
 	select {
 	case <-s.shutdownCh:
 		return s.shutdownErr
+	case <-deadline:
+		return ErrTimeout
```

Comment:
Should this be a separate select statement after this one?

Reply:
Yes, you are right, the shutdownErr should preferred.